### ISSUE TITLE
[agw] Replace hashtable OAI lib to a standard library through map.h interface

### ISSUE THREAD

**Issue by ardzoht** [magma/magma#11190]
### Description

- OAI makes use of its own internal library for hashtable structures. This is a wide-used library being used throughout all MME tasks mostly for the maintenance of both task’s and UE state. There are multiple types available to use, but mostly both uint64 => uint64 and uint64 => object PTR thread-safe mappings are the most used in the codebase.
- This library has shown evidence of being prone to instability across compiler versions, due to its complex memory management with value and key pointer objects: e.g. https://github.com/magma/magma/pull/11865

### Goal
- We want to iteratively migrate OAI’s hashtable and their usage to protobuf's generated object CPP hashmap structs, a starting point to refer can be: `/lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp` or `lte/gateway/c/core/oai/include/map.h`

1. Replace hashtable lib usages in S1AP, NGAP task procedures to `magma::map_s`
2.  Replace hashtable lib usages in SPGW task procedures to `magma::map_s`
3. Replace hashtable lib usages in MME APP task procedures to `magma::map_s`
4. Replace hashtable lib usages in SGW_S8 task procedures to `magma::map_s`

**Comment by ssanadhya** [comment]
@rsarwad , is there any hash table managed by the SGW_S8 task? If yes, please add that to the list of goals for this issue.

**Comment by rsarwad** [comment]
Yes @ssanadhya, SGW_S8 task also includes hash tables. The estimation includes sgw_s8 task

**Comment by shahrahman-fb** [comment]
Closing this issue as part of a repository cleanup effort. This issue has had no activity for over 2 years.

If this issue is still relevant, please reopen it or file a new issue with updated information. We are actively maintaining the project again and want to focus on current, actionable issues.

See: #15833 (Phase 1 cleanup), #15828 (tracking issue)
